### PR TITLE
Doc updates for CodeDOM

### PR DESCRIPTION
**Pull request by danmoseley**
Using the doc porter tool.

Fixes https://github.com/dotnet/runtime/issues/54874

### CODE REVIEW

File: xml/System.CodeDom.Compiler/IndentedTextWriter.xml [hunk 1]
```diff
@@ -1391,10 +1391,10 @@
         <Parameter Name="cancellationToken" Type="System.Threading.CancellationToken" Index="1" FrameworkAlternate="net-6.0" />
       </Parameters>
       <Docs>
-        <param name="buffer">To be added.</param>
-        <param name="cancellationToken">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="buffer">The buffer to write.</param>
+        <param name="cancellationToken">Token for canceling the operation.</param>
+        <summary>Asynchronously writes the specified buffer to the underlying <see cref="T:System.IO.TextWriter" />, inserting tabs at the start of every line.</summary>
```
Comment (buyaa-n) [inline xml/System.CodeDom.Compiler/IndentedTextWriter.xml:1396]:
```suggestion
        <param name="buffer">The character array to write.</param>
        <param name="cancellationToken">Token for canceling the operation.</param>
        <summary>Asynchronously writes the character array to the underlying <see cref="T:System.IO.TextWriter" />, inserting tabs at the start of every line.</summary>
```

File: xml/System.CodeDom.Compiler/IndentedTextWriter.xml [hunk 2]
```diff
@@ -1426,10 +1426,10 @@
         <Parameter Name="cancellationToken" Type="System.Threading.CancellationToken" Index="1" FrameworkAlternate="net-6.0" />
       </Parameters>
       <Docs>
-        <param name="value">To be added.</param>
-        <param name="cancellationToken">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="value">The value to write.</param>
```
Comment (buyaa-n) [inline xml/System.CodeDom.Compiler/IndentedTextWriter.xml:1429]:
```suggestion
        <param name="value">The string, as a string builder to write.</param>
```

File: xml/System.CodeDom.Compiler/IndentedTextWriter.xml [hunk 3]
```diff
@@ -2258,9 +2258,9 @@
         <Parameter Name="value" Type="System.Char" Index="0" FrameworkAlternate="net-6.0" />
       </Parameters>
       <Docs>
-        <param name="value">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="value">The value to write.</param>
```
Comment (buyaa-n) [inline xml/System.CodeDom.Compiler/IndentedTextWriter.xml:2261]:
```suggestion
        <param name="value">The character to write.</param>
```

File: xml/System.CodeDom.Compiler/IndentedTextWriter.xml [hunk 4]
```diff
@@ -2292,9 +2292,9 @@
         <Parameter Name="value" Type="System.String" Index="0" FrameworkAlternate="net-6.0" />
       </Parameters>
       <Docs>
-        <param name="value">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="value">The value to write.</param>
```
Comment (buyaa-n) [inline xml/System.CodeDom.Compiler/IndentedTextWriter.xml:2295]:
```suggestion
        <param name="value">The string to write.</param>
```

File: xml/System.CodeDom.Compiler/IndentedTextWriter.xml [hunk 5]
```diff
@@ -2361,10 +2361,10 @@
         <Parameter Name="cancellationToken" Type="System.Threading.CancellationToken" Index="1" FrameworkAlternate="net-6.0" />
       </Parameters>
       <Docs>
-        <param name="value">To be added.</param>
-        <param name="cancellationToken">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="value">The text to write.</param>
```
Comment (buyaa-n) [inline xml/System.CodeDom.Compiler/IndentedTextWriter.xml:2364]:
```suggestion
        <param name="value">The string, as a string builder to write.</param>
```

File: xml/System.CodeDom.Compiler/IndentedTextWriter.xml [hunk 6]
```diff
@@ -2398,11 +2398,11 @@
         <Parameter Name="count" Type="System.Int32" Index="2" FrameworkAlternate="net-6.0" />
       </Parameters>
       <Docs>
-        <param name="buffer">To be added.</param>
-        <param name="index">To be added.</param>
-        <param name="count">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="buffer">The buffer to write from.</param>
```
Comment (buyaa-n) [inline xml/System.CodeDom.Compiler/IndentedTextWriter.xml:2401]:
```suggestion
        <param name="buffer">The character array to write data from.</param>
```

File: xml/System.CodeDom.Compiler/IndentedTextWriter.xml [hunk 7]
```diff
@@ -2492,9 +2492,9 @@
         <Parameter Name="s" Type="System.String" Index="0" FrameworkAlternate="net-6.0" />
       </Parameters>
       <Docs>
-        <param name="s">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="s">The text to write.</param>
+        <summary>Asynchronously writes the specified text to the underlying <see cref="T:System.IO.TextWriter" /> without inserting tabs.</summary>
```
Comment (buyaa-n) [inline xml/System.CodeDom.Compiler/IndentedTextWriter.xml:2496]:
```suggestion
        <param name="s">The string to write.</param>
        <summary>Asynchronously writes the specified string to the underlying <see cref="T:System.IO.TextWriter" /> without inserting tabs.</summary>
```

**Review by buyaa-n — APPROVED** [review]
Added some NIT comments to make the parameter definitions the same as in the non-async doc. Overall LGTM, thank you!

File: xml/System.CodeDom.Compiler/IndentedTextWriter.xml (hunk 1 above)
Comment (danmoseley) [reply xml/System.CodeDom.Compiler/IndentedTextWriter.xml:1396]: I used 'characters' as it's not necessarily an array.

File: xml/System.CodeDom.Compiler/IndentedTextWriter.xml (hunk 6 above)
Comment (danmoseley) [reply xml/System.CodeDom.Compiler/IndentedTextWriter.xml:2401]: I decided to leave some of these -- I don't think the purpose of the intellisense text is to say what the type is -- the tool (VS etc) shows that. It's to say what it's used for, or what it contains. As it's a text writer, any buffer to write from is going to be a string/char/span<char>/stringbuilder.

Comment (buyaa-n) [reply xml/System.CodeDom.Compiler/IndentedTextWriter.xml:2401]: Sounds good

**Comment by danmoseley** [comment]
Updated code and then updated this PR.

We definitely want to get to a point where PR's like this get no comments, because the review was all done with the code commit.

**Pull request merged**